vault/identity_store_oidc_provider_test.go

```diff
@@ -1317,17 +1556,19 @@ func testAssignmentReq(s logical.Storage, entityID, groupID string) *logical.Req
 	}
 }
 
-func testClientReq(s logical.Storage) *logical.Request {
+func testClientReq(s logical.Storage, authorizationCode, clientCredentials bool) *logical.Request {
 	return &logical.Request{
 		Storage:   s,
 		Path:      "oidc/client/test-client",
 		Operation: logical.CreateOperation,
 		Data: map[string]interface{}{
-			"key":              "test-key",
-			"redirect_uris":    []string{"https://localhost:8251/callback"},
-			"assignments":      []string{"test-assignment"},
-			"id_token_ttl":     "24h",
-			"access_token_ttl": "24h",
+			"key":                "test-key",
+			"redirect_uris":      []string{"https://localhost:8251/callback"},
+			"assignments":        []string{"test-assignment"},
+			"id_token_ttl":       "24h",
+			"access_token_ttl":   "24h",
+			"client_credentials": clientCredentials,
+			"authorization_code": authorizationCode,
```

Comment:
Rather than directly setting these, I would like to see verification of the default settings.  Please leave these unset when you initially create a data map, and only set the values if they are not the default setting.

Reply:
I have implemented this but it has one Downside. In `TestOIDC_Path_OIDC_Authorize` the clients are not reset or deleted. This makes perfect sense for testing speed. 

But when I want to test `invalid authorize provider does not allow authorization code flow` `authorization_code` is set to false the state is never set back to the default value. 

I put this test at the end so the test is working.
And noted the reason at the test.